examples/instance_refresh/main.tf

```diff
@@ -0,0 +1,259 @@
+      ]
+    }
+  )
+}
+
+resource "aws_cloudwatch_event_target" "node_term_event_target" {
+  rule      = aws_cloudwatch_event_rule.node_term_event_rule.name
+  target_id = "ANTHandler"
+  arn       = module.node_term_sqs.sqs_queue_arn
+}
+
+module "node_term_role" {
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "4.1.0"
+  create_role                   = true
+  role_description              = "IRSA role for ANTH, cluster ${local.cluster_name}"
+  role_name_prefix              = local.cluster_name
+  provider_url                  = replace(module.eks.cluster_oidc_issuer_url, "https://", "")
+  role_policy_arns              = [aws_iam_policy.node_term.arn]
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${var.namespace}:${var.serviceaccount}"]
+}
+
+resource "null_resource" "helm_install" {
+  provisioner "local-exec" {
+    environment = {
+      USE_SUDO         = "false"
+      HELM_INSTALL_DIR = "${path.root}/bin"
+    }
+    command = <<EOT
+mkdir -p ${path.root}/bin
+curl -fsSL -o ${path.root}/bin/get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+chmod +x ${path.root}/bin/get_helm.sh
+${path.root}/bin/get_helm.sh
+EOT
+  }
+}
+
+resource "null_resource" "helm_add_repo" {
+  depends_on = [
+    null_resource.helm_install,
+  ]
+
+  triggers = {
+    version = var.aws_node_termination_handler_chart_version
+  }
+
+  provisioner "local-exec" {
+    environment = {
+      XDG_CACHE_HOME  = abspath("${path.root}/.cache")
+      XDG_CONFIG_HOME = abspath(path.root)
+      XDG_DATA_HOME   = abspath(path.root)
+    }
+    command = "${path.root}/bin/helm repo add --force-update eks https://aws.github.io/eks-charts"
+  }
+}
+
+resource "null_resource" "helm_install_chart" {
+  depends_on = [
+    module.eks,
+    module.node_term_role,
+    null_resource.helm_add_repo,
+  ]
+
+  triggers = {
+    version  = var.aws_node_termination_handler_chart_version
+    role_arn = module.node_term_role.iam_role_arn
+  }
+
+  provisioner "local-exec" {
+    environment = {
+      KUBECONFIG      = "${path.root}/${module.eks.kubeconfig_filename}"
+      XDG_CACHE_HOME  = abspath("${path.root}/.cache")
+      XDG_CONFIG_HOME = abspath(path.root)
+      XDG_DATA_HOME   = abspath(path.root)
+    }
+    command = <<EOT
+chmod 0600 "$KUBECONFIG"
+${path.root}/bin/helm upgrade --install aws-node-termination-handler \
+  --wait \
+  --namespace ${var.namespace} \
+  --set serviceAccount.name=${var.serviceaccount} \
+  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${self.triggers.role_arn} \
+  --set enableSqsTerminationDraining=true \
+  --set queueURL=${module.node_term_sqs.sqs_queue_id} \
+  --set logLevel=DEBUG \
+  --set awsRegion=${var.region} \
+  --version=${self.triggers.version} \
+  eks/aws-node-termination-handler
+EOT
+  }
+}
+
+resource "aws_autoscaling_lifecycle_hook" "node_term" {
```

Comment:
If you create this lifecycle hook, you don't need to set worker_create_initial_lifecycle_hook to true.

Reply:
fixed.

Reply:
@bashims did you read this https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group ?

![image](https://user-images.githubusercontent.com/550783/118511252-01b7c280-b732-11eb-8773-73138c19fc3f.png)

 I was wondering if it's not better to use initial_lifecycle_hooks https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf#L37.

@bashims @stevehipwell @antonbabenko any thoughts ?

Reply:
I think @bryantbiggs knows better about this.

Reply:
I tried the `initial_lifecycle_hook` and there is an issue on destroy. Basically the ANTH cannot complete the lifecycle action since it is terminated before the ASG is destroyed. This causes Terraform to timeout waiting for the ASG's destruction.

Reply:
ohh thanks. Can you please add this as a comment in the example. It'll avoid asking the same question again and again ^^.

Reply:
Absolutely.